dll/win32/dbgeng/dbgeng.c

```diff
@@ -0,0 +1,4133 @@
+}
+
+static HRESULT debug_target_return_string(const char *str, char *buffer, unsigned int buffer_size,
+        unsigned int *size)
+{
+    unsigned int len = strlen(str), dst_len;
+
+    if (size)
+        *size = len + 1;
+
+    if (buffer && buffer_size)
+    {
+        dst_len = min(len, buffer_size - 1);
+        if (dst_len)
+            memcpy(buffer, str, dst_len);
+        buffer[dst_len] = 0;
+    }
+
+    return len < buffer_size ? S_OK : S_FALSE;
+}
+
+static WORD debug_target_get_module_machine(struct target_process *target, HMODULE module)
+{
+    IMAGE_DOS_HEADER dos = { 0 };
+    WORD machine = 0;
+
+    ReadProcessMemory(target->handle, module, &dos, sizeof(dos), NULL);
+    if (dos.e_magic == IMAGE_DOS_SIGNATURE)
+    {
+        ReadProcessMemory(target->handle, (const char *)module + dos.e_lfanew + 4 /* PE signature */, &machine,
+                sizeof(machine), NULL);
+    }
+
+    return machine;
+}
+
+static DWORD debug_target_get_module_timestamp(struct target_process *target, HMODULE module)
+{
+    IMAGE_DOS_HEADER dos = { 0 };
+    DWORD timestamp = 0;
+
+    ReadProcessMemory(target->handle, module, &dos, sizeof(dos), NULL);
+    if (dos.e_magic == IMAGE_DOS_SIGNATURE)
+    {
+        ReadProcessMemory(target->handle, (const char *)module + dos.e_lfanew + 4 /* PE signature */ +
+                FIELD_OFFSET(IMAGE_FILE_HEADER, TimeDateStamp), &timestamp, sizeof(timestamp), NULL);
+    }
+
+    return timestamp;
+}
+
+static HRESULT debug_target_init_modules_info(struct target_process *target)
+{
+    unsigned int i, count;
+    HMODULE *modules;
+    MODULEINFO info;
+    DWORD needed;
+
+    if (target->modules.initialized)
+        return S_OK;
+
+    if (!target->handle)
+        return E_UNEXPECTED;
+
+    needed = 0;
+    EnumProcessModules(target->handle, NULL, 0, &needed);
+    if (!needed)
+        return E_FAIL;
+
+    count = needed / sizeof(HMODULE);
+
+    if (!(modules = heap_alloc(count * sizeof(*modules))))
+        return E_OUTOFMEMORY;
+
+    if (!(target->modules.info = heap_alloc_zero(count * sizeof(*target->modules.info))))
+    {
+        heap_free(modules);
+        return E_OUTOFMEMORY;
+    }
+
+    if (EnumProcessModules(target->handle, modules, count * sizeof(*modules), &needed))
+    {
+        for (i = 0; i < count; ++i)
+        {
+            if (!GetModuleInformation(target->handle, modules[i], &info, sizeof(info)))
+            {
+                WARN("Failed to get module information, error %d.\n", GetLastError());
+                continue;
+            }
+
+            target->modules.info[i].params.Base = (ULONG_PTR)info.lpBaseOfDll;
+            target->modules.info[i].params.Size = info.SizeOfImage;
+            target->modules.info[i].params.TimeDateStamp = debug_target_get_module_timestamp(target, modules[i]);
+
+            GetModuleFileNameExA(target->handle, modules[i], target->modules.info[i].image_name,
+#ifdef __REACTOS__
```

Comment:
You should define `__WINESRC__` in CMakeLists.txt. That should make `ARRAY_SIZE` available here.

Reply:
Fixed. Thanks @ThFabba. I missed that.
Still compilable for me.